tests/integration/pilot/common/routing.go

```diff
@@ -2775,6 +2776,48 @@ type vmCase struct {
 	host string
 }
 
+func ExternalServiceCases(t TrafficContext) {
+	sidecarScope := fmt.Sprintf(`
+apiVersion: networking.istio.io/v1alpha3
+kind: Sidecar
+metadata:
+  name: restrict-external-service
+  namespace: %s
+spec:
+  workloadSelector:
+    labels:
+      app: a
+  outboundTrafficPolicy:
+    mode: "REGISTRY_ONLY"
+	`, t.Apps.EchoNamespace.Namespace.Name())
+
+	deployment.DeployExternalServiceEntry(t.ConfigIstio(), t.Apps.Namespace, t.Apps.External.Namespace).
+		ApplyOrFail(t, apply.CleanupConditionally)
+
+	client := t.Apps.A[0]
+	for _, port := range []echo.Port{ports.HTTP, ports.HTTPS} {
+		var s scheme.Instance
+		switch strings.ToLower(port.Name) {
+		case "https":
+			s = scheme.HTTPS
+		default:
+			s = scheme.HTTP
+		}
+		t.RunTraffic(TrafficTestCase{
+			name:   "external-service-" + port.Name,
+			call:   client.CallOrFail,
```

Comment:
I donot think this test can verify ipv6

Reply:
Yeah I'm not 100% sure how to force the ipv6 address with the echo framework. Any ideas?

Reply:
Hi @keithmattix, may change the code for configuration of calling `deployment.DeployExternalServiceEntry` ? In this function, it will config a service entry, and you can set the endpoint addresses for it, such as :
For IPv6: 
```
endpoints:
  - address: 1111:2222:3333:4444::0:1
  - address: 1111:2222:3333:4444::0:2
```

For Dual Stack:
```
endpoints:
  - address: 1.2.3.4
  - address: 1111:2222:3333:4444::0:2
```

Reply:
If we add addresses though, the Envoy config will listen on those addresses instead of the wildcard. We can't set addresses or endpoints if we want to create the conditions for the bug

Reply:
I wrote a unit test instead; I don't see a way to write an integration test for this right now

Reply:
Hi @keithmattix, you are right, we need to add `curl -6` in the echo framework in the integration test, I think @kfaseela is working on this as well. it's okay from my side. Thanks for your PR!